### PR TITLE
Added support for component injection into custom toasts

### PR DESCRIPTION
**Pull request by CiprianDraghici**
### Feature

This feature allows custom components to be injected in the same container with the transactions' toasts.

### CODE REVIEW

**Review by ioandanpopa — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+- [Custom toast extensions](https://github.com/multiversx/mx-sdk-dapp/pull/848)
```
Comment (arhtudormorar) [inline CHANGELOG.md:10]: A bit better description starting with Added ?

File: src/UI/TransactionsToastList/TransactionsToastList.tsx [hunk 2]
```diff
@@ -126,14 +131,20 @@ export const TransactionsToastList = ({
...
+  const customToastsList = customToasts.map((props) => {
+    const CustomComponent =
+      getRegisteredCustomIconComponents(props.toastId) ?? null;
+
+    return (
+      <CustomToast
+        key={props.toastId}
+        {...props}
+        component={CustomComponent as never}
+        onDelete={() => handleDeleteCustomToast(props.toastId)}
```
Comment (arhtudormorar) [inline src/UI/TransactionsToastList/TransactionsToastList.tsx:143]: declare above

File: src/UI/TransactionsToastList/TransactionsToastList.tsx [hunk 3]
```diff
@@ -126,14 +131,20 @@ export const TransactionsToastList = ({
...
-  ));
+  const customToastsList = customToasts.map((props) => {
+    const CustomComponent =
+      getRegisteredCustomIconComponents(props.toastId) ?? null;
+
+    return (
+      <CustomToast
+        key={props.toastId}
+        {...props}
+        component={CustomComponent as never}
```
Comment (arhtudormorar) [inline src/UI/TransactionsToastList/TransactionsToastList.tsx:142]: is as never needed?

File: src/UI/TransactionsToastList/TransactionsToastList.tsx [hunk 4]
```diff
@@ -126,14 +131,20 @@ export const TransactionsToastList = ({
...
-    />
-  ));
+  const customToastsList = customToasts.map((props) => {
+    const CustomComponent =
+      getRegisteredCustomIconComponents(props.toastId) ?? null;
+
+    return (
+      <CustomToast
+        key={props.toastId}
+        {...props}
```
Comment (arhtudormorar) [inline src/UI/TransactionsToastList/TransactionsToastList.tsx:141]: declare props with strong type imported above. Destructuring hides props from TS

File: src/UI/TransactionsToastList/components/CustomToast/components/CustomComponentToast.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      id={toastId}
+      className={classNames(
+        wrapperStyles.toasts,
+        wrapperStyles.toastWrapper,
+        className
+      )}
+    >
+      <>
+        {closeButton}
+        {component ? component() : message}
```
Comment (arhtudormorar) [inline src/UI/TransactionsToastList/components/CustomToast/components/CustomComponentToast.tsx:28]: I think `component()` is not correct. Best to use TSX

**Pull request merged**